### ISSUE TITLE
Code splitting in Typescript

### ISSUE THREAD

**Issue by ccheraa** [parcel-bundler/website#546]
The [documentation page](https://parceljs.org/code_splitting.html) does not mention "Typescript" at all, and says that code splitting works automatically on dynamic import, yet I still can't to get it to work.
I found a couple of issues talking about it ([#1547](https://github.com/parcel-bundler/parcel/issues/1547) and [#2331](https://github.com/parcel-bundler/parcel/issues/2331)) yet couldn't make it work.
So I think it's better if we have the documentation mentions what extra work needs to be done to get code splitting to work when using Typescript.

**Comment by mischnic** [comment]
Could you give an example where it doesn't work in your case?

**Comment by 190n** [comment]
I was able to get code splitting to work (in [a bare-bones project](https://github.com/190n/parcel-typescript-code-splitting) and an actual app where I had earlier been trying to make it work) using @reflog's `tsconfig.json` from [#1547](https://github.com/parcel-bundler/parcel/issues/1547#issuecomment-397630691). But it didn't work until I cleared Parcel's cache (`rm -r .cache` from project root).

tsconfig.json:

```json
{
    "compilerOptions": {
        "target": "ES2018",
        "sourceMap": true,
        "module": "ESNext",
        "jsx": "preserve",
        "lib": ["ESNext", "DOM"]
    }
}

```

Anecdotally, Parcel 2 seems to handle it fine without the tsconfig.json, but I will probably continue to use the workaround as I'm not ready to migrate yet (I also have other TypeScript customizations, so it's not like I'm adding a whole config file to fix one issue).
